ui_testing/pages/testplan_page.py

```diff
@@ -44,17 +45,19 @@ def set_test_plan(self, name='', random=False):
         self.base_selenium.set_text_in_drop_down(ng_select_element='test_plan:test_plan', text=name)
         return name
 
-    def set_test_unit(self, test_unit='', **kwargs):
+    def set_test_unit(self, test_unit='', add_limits=True, **kwargs):
         self.base_selenium.click('test_plan:next')
         self.base_selenium.click('test_plan:add_test_units')
-        self.base_selenium.select_item_from_drop_down(element='test_plan:test_units', item_text=test_unit)
+        testunit_added_result = self.base_selenium.select_item_from_drop_down(element='test_plan:test_units', item_text=test_unit)
+        if testunit_added_result == False:
+            return False
         self.base_selenium.click('test_plan:add')
-        if 'upper' in kwargs:
+        if add_limits and 'upper' in kwargs:
             self.base_selenium.LOGGER.info(' set upper : {}'.format(kwargs['upper']))
             elems = self.base_selenium.find_elements('general:col_6')
             upper = self.base_selenium.find_element_in_element(source=elems[4], destination_element='general:input')
             upper.send_keys(kwargs['upper'])
-        if 'lower' in kwargs:
+        if add_limits and  'lower' in kwargs:
```

Comment:
no need to do that as the old code only execute if the 'lower' in kwatgs

Reply:
done